src/overlays/actors/ovl_Bg_Ikana_Dharma/z_bg_ikana_dharma.c

```diff
@@ -28,47 +34,229 @@ const ActorInit Bg_Ikana_Dharma_InitVars = {
+        OC1_NONE,
+        OC2_NONE,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000100, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_NONE,
+    },
     { 98, 10, 25, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80BECAFC[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 320, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 320, ICHAIN_CONTINUE),
     ICHAIN_F32_DIV1000(gravity, -1100, ICHAIN_STOP),
 };
 
-#endif
+static BgIkanaDharma* g_firstHitBgIkanaDharma;
 
-extern ColliderCylinderInit D_80BECAD0;
-extern InitChainEntry D_80BECAFC[];
+void BgIkanaDharma_CreateParticles(BgIkanaDharma* this, GlobalContext* globalCtx) {
+    u32 pad;
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f accel;
+    f32 speed;
+    f32 dirX;
+    f32 initialY;
+    f32 dirZ;
+    f32 initialRadius;
+    s16 angle;
+    s32 i;
 
-extern UNK_TYPE D_060008C8;
-extern UNK_TYPE D_06000C50;
+    initialRadius = this->dyna.actor.scale.x * 200.0f;
+    initialY = this->dyna.actor.scale.y * 50.0f;
+    for (i = 0; i != 4; i++) {
+        speed = (Rand_ZeroOne() * 5.0f) + 5.0f;
+        angle = ((u32)Rand_Next() >> 0x12) + this->dyna.actor.world.rot.y + 0x6000;
+        dirX = Math_SinS(angle);
+        dirZ = Math_CosS(angle);
+        pos.x = (dirX * initialRadius) + this->dyna.actor.world.pos.x;
+        pos.y = this->dyna.actor.world.pos.y + initialY;
+        pos.z = (dirZ * initialRadius) + this->dyna.actor.world.pos.z;
+        velocity.x = speed * dirX;
+        velocity.y = Rand_ZeroOne() + 0.5f;
+        velocity.z = speed * dirZ;
+        accel.x = velocity.x * -0.05f;
+        accel.y = -0.15f;
+        accel.z = velocity.z * -0.05f;
+        EffectSsKiraKira_SpawnSmallYellow(globalCtx, &pos, &velocity, &accel);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC240.s")
+void BgIkanaDharma_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s16 params;
+    BgIkanaDharma* this = THIS;
+    f32 segmentY;
+    GlobalContext* globalCtx2 = globalCtx;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/BgIkanaDharma_Init.s")
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->dyna.actor.scale.x = 0.3f;
+    this->dyna.actor.scale.y = 0.1f;
+    this->dyna.actor.scale.z = 0.3f;
+    DynaPolyActor_Init(&this->dyna, 0);
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_ikana_obj_Colheader_000C50);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx2, &this->collider, &this->dyna.actor, &sCylinderInit);
+    i = 0;
+    params = this->dyna.actor.params;
+    if (((params >> 5) & 1) == 0) {
+        segmentY = this->dyna.actor.world.pos.y;
+        if (i < (params & 0xF)) {
```

Comment:
I'm suspicious that these params are an actual local temp, unless its modified they are usually just in-line
(some actors clear some of their data from params so params from that point can just be one variable re-used in the actor, like a flag is taken for init but cleared from params with `& 0xFF` after pulled)

Either way though, Kenix generally wants them to be wrapped in a macro like you did with the switch_flags, even if you dont know what these params do yet he generally wants them wrapped with something like `GET_BGIKANADHARMA_100` if you don't know what to name them
(although at a glance kinda looks like & 0xF is how many blocks tall it spawns with?)